jerry-core/ecma/operations/ecma-objects.c

```diff
@@ -138,6 +142,43 @@ ecma_op_object_get_own_property (ecma_object_t *object_p, /**< the object */
       }
       break;
     }
+#ifndef CONFIG_DISABLE_TYPEDARRAY_BUILTIN
+    case ECMA_OBJECT_TYPE_PSEUDO_ARRAY:
+    {
+      /* ES2015 9.4.5.1 */
+      if (ecma_is_typedarray (ecma_make_object_value (object_p)))
+      {
+        if (ECMA_STRING_GET_CONTAINER (property_name_p) == ECMA_STRING_CONTAINER_UINT32_IN_DESC)
+        {
+          ecma_value_t value = ecma_op_typedarray_get_index_prop (object_p, property_name_p->u.uint32_number);
+          if (!ecma_is_value_undefined (value))
+          {
+            property_ref_p->virtual_value = value;
+
+            return ECMA_PROPERTY_ENUMERABLE_WRITABLE | ECMA_PROPERTY_TYPE_VIRTUAL;
+          }
+          else
+          {
+            return ECMA_PROPERTY_TYPE_NOT_FOUND_AND_STOP;
+          }
+        }
+
+        ecma_number_t num = ecma_string_to_number (property_name_p);
+        ecma_string_t *num_to_str = ecma_new_ecma_string_from_number (num);
```

Comment:
I think at some point we should optimize this by a function which checks the string is a number (but does not decode it). But not now, this patch is big enough.